common/spring-cloud-stream-modules-common-configuration/src/main/java/org/springframework/cloud/stream/module/file/remote/AbstractRemoteFileSourceProperties.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.stream.module.file.remote;
+
+import java.io.File;
+
+/**
+ * @author David Turanski
+ * @author Gary Russell
+ *
+ */
+public abstract class AbstractRemoteFileSourceProperties {
+
+	/**
+	 * The remote FTP directory, "/" by default.
+	 */
+	private String remoteDir = "/";
+
+	/**
+	 * Set to true to delete remote files after successful transfer, false by default.
+	 */
+	private boolean deleteRemoteFiles = false;
+
+	/**
+	 * The local directory to use for file transfers, System.getProperty("java.io.tmpdir") + "/xd/ftp" by default.
+	 */
+	private File localDir = new File(System.getProperty("java.io.tmpdir") + "/xd/ftp");
+
+	/**
+	 * Set to true to create the local directory if it does not exist, true by default
+	 */
+	private boolean autoCreateLocalDir = true;
+
+	/**
+	 * The suffix to use while the transfer is in progress, ".tmp" by default.
+	 */
+	private String tmpFileSuffix = ".tmp";
+
+	/**
+	 * A filter pattern to match the names of files to transfer, no filter by default.
+	 */
+	private String filenamePattern;
```

Comment:
Is this mutually exclusive with `filenameRegex`? If so, exclusion should be implemented by jsr303 validation